src/wp-includes/class-wp-scripts.php

```diff
@@ -714,6 +900,208 @@ public function in_default_dir( $src ) {
+				_doing_it_wrong(
+					__METHOD__,
+					sprintf(
+						/* translators: 1: $strategy, 2: $handle */
+						__( 'Cannot supply a strategy `%1$s` for script `%2$s` because it does not have a `src` value.' ),
+						$value,
+						$handle
+					),
+					'6.3.0'
+				);
+				return false;
+			}
+		}
+		return parent::add_data( $handle, $key, $value );
+	}
+
+	/**
+	 * Checks all handles for any delayed inline scripts.
+	 *
+	 * @since 6.3.0
+	 * @see WP_Scripts::should_delay_inline_script()
+	 *
+	 * @param string[] $handles Handles to check.
+	 * @return bool True if the inline script present, otherwise false.
+	 */
+	public function has_delayed_inline_script( array $handles ) {
+		foreach ( $handles as $handle ) {
+			foreach ( array( 'before', 'after' ) as $position ) {
+				if (
+					$this->get_data( $handle, $position ) &&
+					$this->should_delay_inline_script( $handle, $position )
+				) {
+					return true;
+				}
+			}
+		}
+		return false;
+	}
+
+	/**
+	 * Gets all dependents of a script.
+	 *
+	 * @since 6.3.0
+	 *
+	 * @param string $handle The script handle.
+	 * @return string[] Script handles.
+	 */
+	private function get_dependents( $handle ) {
+		// Check if dependents map for the handle in question is present. If so, use it.
+		if ( array_key_exists( $handle, $this->dependents_map ) ) {
+			return $this->dependents_map[ $handle ];
+		}
+
+		$dependents = array();
+
+		// Iterate over all registered scripts, finding dependents of the script passed to this method.
+		foreach ( $this->registered as $registered_handle => $args ) {
+			if ( in_array( $handle, $args->deps, true ) ) {
+				$dependents[] = $registered_handle;
+			}
+		}
+
+		// Add the handles dependents to the map to ease future lookups.
+		$this->dependents_map[ $handle ] = $dependents;
+
+		return $dependents;
+	}
+
+	/**
+	 * Checks if the strategy passed is a valid delayed (non-blocking) strategy.
+	 *
+	 * @since 6.3.0
+	 *
+	 * @param string $strategy The strategy to check.
+	 * @return bool True if $strategy is one of the delayed strategies, otherwise false.
+	 */
+	private function is_delayed_strategy( $strategy ) {
+		return in_array(
+			$strategy,
+			$this->delayed_strategies,
+			true
+		);
+	}
+
+	/**
+	 * Checks if all of a script's dependents are delayed, which is required to maintain execution order.
+	 *
+	 * @since 6.3.0
+	 *
+	 * @param string              $handle     The script handle.
+	 * @param bool                $async_only Whether to limit to async strategy only.
+	 * @param array<string, true> $checked    Optional. An array of already checked script handles, used to avoid recursive loops.
+	 * @return bool True if all dependents are delayed, false otherwise.
+	 */
+	private function has_only_delayed_dependents( $handle, $async_only = false, $checked = array() ) {
+		// If this node was already checked, this script can be delayed and the branch ends.
+		if ( array_key_exists( $handle, $checked ) ) {
```

Comment:
https://github.com/WordPress/wordpress-develop/pull/4391/files?file-filters%5B%5D=.php&show-viewed-files=true#r1232052943

Reply:
Done in https://github.com/10up/wordpress-develop/pull/80